### DISCUSSION TITLE
2FA lost google authenticator

### DISCUSSION THREAD

**Discussion by joaopa24** [community/community#118977] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

I would like to know the community's opinion about my case, because support doesn't respond and from what I read on other forums they won't help me. I lost my 2FA code because I uninstalled the Google Authenticator app by accident, I discovered that it didn't store the codes in the cloud (it only started recently), github had recommended this app to me and didn't say anything about this serious security flaw, the account has the my name, my photo, the sessions she was logged in to are from my hometown and the city where I go to college, the projects that I still have access to are from courses I took and I can prove it, I have private projects that are work specific college. I'm saying this because I lost my recovery codes, 2FA recovery is not recognizing my PC as a trusted device and I don't have SSH, what do you think of my situation, is there light at the end of the tunnel or should I now migrate platforms?

**Comment by jvmhonorato** [comment]
Do you have access to your linked account?
If yes, read the following documentation:

https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-personal-account/unlinking-your-email-address-from-a-locked-account

**Reply by joaopa24** [reply]
This is to unlink the account, I'm afraid to do this because I don't remember having configured my email to be public, so commits cannot link to my new account.

**Reply by joaopa24** [reply]
@Honorato-Dev Is the probability of them giving me my account back zero?

**Reply by jvmhonorato** [reply]
Without the recovery code it's difficult, try opening a ticket in the contacts tab explaining your situation.

https://support.github.com/request

**Comment by kalebers** [comment]
Hello, 

You can try to send a ticket to GitHub explaining your situation with ***_much details possible_***. You may have a possibility to recover your account.  Give it a try: https://support.github.com/request

**Reply by joaopa24** [reply]
@kalebers I told all the details, cities in which I log in, my name and photo are on the profile, projects I did are from a specific course in my country, the modules needed for the projects, I said everything..., but it was in email separated in my reply from them, will this interfere?

**Reply by kalebers** [reply]
Not at all, had similar issue in another platform, the process was a bit slow.  In my case, I sent different tickets and information directly to their email, but worked for me. A separated email won't change anything.

Now it's up to GitHub confirm your information and do their turn. Hope it works for you too!

**Reply by joaopa24** [reply]
@kalebers The github documentation says that in situations like this, if a person loses the recovery methods, they cannot prove identity, even the github profile showing a person, such as name and photo, I will wait longer

**Reply by kalebers** [reply]
Hi @joknz21!

I checked more on the internet, there's a big chance that without the recovery methods you won't succeed to recover you account. 

In that case I personally would not rely on the little chance that GitHub may have to help you. 

I would suggest to search on your device somewhere in the app saving files directory for a code backup. Usually, when the app does not stores the codes online, there's a file somewhere that may be possible to use to see the codes. I used Authy for many years, I found out that there is a backup on my phone for some codes. Any help counts, hope that works for you.

Keep update here, want to see if it'll work for you. Best wishes for you!

**Reply by joaopa24** [reply]
I used Google Authenticator for this, surprisingly the version before the new one in 2023 didn't have cloud saves or logs, I thought it was something very basic for the company to have and I didn't even worry.
I tried to find files like this but I was unsuccessful

**Reply by kalebers** [reply]
That's unfortunate. Hope that you gain access again for you account, but as mentioned before, without any recovery or way to prove won't be easy or possible to do it. Really hope that works for you. Keep posted  here, may help others as well :)

**Reply by joaopa24** [reply]
Dude, I need your opinion, like, I lost this account, but I have public and private repositories, which I also think are stored on the computer, but I was trying to recover the account due to the commit history, because it shows that I had been studying since 2020 Practically, do you think that contribution time is highly valued in a profile?

**Reply by kalebers** [reply]
Well, one of the things that makes GitHub a good platform is regarding the history of projects you have made. 
So yes, it has a  weight on your profile. It's always good to have a place that you can share about your work and show your progress.

BUT DON'T PANIC!

If you still have your projects stored locally, using Git to push your projects you can set past dates for it :) 
Follow this comment hint and you be able to do it :) 
- https://stackoverflow.com/questions/3895453/how-do-i-make-a-git-commit-in-the-past#:~:text=Commit%20changes%20using%20git%20commit,leave%20the%20message%20as%2Dis.

Hoping that you don't need to start over, but IF it happens, you be able to do it with this Git commands.

Any other doubts, I'll try to help you :)

**Comment by biggy-toothberg** [comment]
Hello, I'm facing the same issue now. About a month ago, my phone, which had the authenticator app, flew out of a roller coaster (dont ask how..). However, until now, I've been able to log in without it. Now, after a PC reset, it's asking for my authenticator, which I unfortunately no longer have. I also never set up an SSH connection on my devices or anything like that.

The problem I'm facing now is that my GitHub account is quite important to me. Among other things, there's a project with well over 1K stars and 350+ forks running on it. And the community involved there is growing steadily. But if I no longer have access to this account, I won't be able to accept new PRs or make commits myself.

What should I do now? I've been in touch with support for a few days but haven't really made any progress yet. I can provide my school email, phone number, and details of private projects as proof. I also still have the same IP address... 

really annoying.

**Reply by kalebers** [reply]
Hello @Bahdman1998,

As mentioned before, without any recovery methods (recovery keys from GitHub for example,)  is quite difficult to gain access or prove anything. 
Although, you may have a chance to recover it by the Support, as you already did. 

My advice is: **take advantage of all information you have from your account and send it all very detailed and very well explained**.

Hope it works well for you! :) 
Any doubts, I'll try to help!

**Reply by biggy-toothberg** [reply]
Thank you, I just hope I get lucky. I also find it rather 'meh' that you can submit everything and they still won't give you the account back.

**Reply by kalebers** [reply]
Hope so, but I personally think that is possible to do a 2FA reset sending a link to confirm on your sign in email. Let's see, wish you the best! :)

**Reply by biggy-toothberg** [reply]
I just want to complain once again about the support system at GitHub. You're guided through by a rather lacking support bot or you create a ticket. When you create a ticket, it takes an incredibly long time to receive a response that seems like a copy and paste from the bot.

Then you respond logically to this support email and wait again (for me, more than 48 hours, and I'm still waiting).

And then I don't understand why it has to be so cumbersome to address a user's concerns. I don't just use this GitHub account to store my personal stuff in a repo, but also to work on public projects and support other projects. Additionally, I use this account as part of my resume. And now I'm dependent on sluggish support.

Btw greetings from Germany :)

**Reply by kalebers** [reply]
Hi! Greetings from Germany too!

For some weird reason Support systems for so many places are really terrible, but I was surprised with GitHub in this regard, could be more friendly. I personally think that adding bots to perform tasks sometimes are not effective, in the end of the triage a person will look for your issue anyway. I think if you need support, at least the bot redirect for a free person that could help you at that moment. Agree with you. 

Hope that works for you, keep us posted :)

**Reply by biggy-toothberg** [reply]
I would be interested to know if the creator of the thread has been assisted up to now. 

Kind regards :)

**Reply by biggy-toothberg** [reply]
I've been waiting for over 3 weeks for a single response from support. Starting to believe they're just ignoring me. NEVER seen such awful support. It's like they're literally shitting on the users...

**Reply by kalebers** [reply]
Hi, usually it takes a bit of time to process every detail about this cases, but unfortunately I have to agree that it takes too much time for something crucial. 
I would suggest to wait for maybe a week, and if you don't hear anything, send a ticekt again, wait a little longer just to avoid making spam in case you don't have any reply. 

keep posted, hope you can find a solution for your problem! Any doubt, just let me know and I'll try to help :)

**Comment by leofun01** [comment]
> should I now migrate platforms?

Only if you want to migrate or if you have no choice.
- [codeberg.org](https://codeberg.org/)
- [sourceforge.net](https://sourceforge.net/)
- [sr.ht](https://git.sr.ht/)
- [notabug.org](https://notabug.org/)
- [gitea.com](https://gitea.com/)
- [gitlab.com](https://about.gitlab.com/)
- [disroot.org](https://git.disroot.org/)
- [savannah.gnu.org](https://savannah.gnu.org/)
